corehq/apps/locations/models.py

```diff
@@ -157,23 +157,32 @@ def expand_from_root(self, value):
     def commtrack_enabled(self):
         return Domain.get_by_name(self.domain).commtrack_enabled
 
-    def _populate_stock_levels(self, config):
+    def _populate_stock_levels(self, config, update_fields=None):
         self.emergency_level = config.emergency_level
         self.understock_threshold = config.understock_threshold
         self.overstock_threshold = config.overstock_threshold
+        if update_fields:
+            update_fields.append('emergency_level')
+            update_fields.append('understock_threshold')
+            update_fields.append('overstock_threshold')
 
     def save(self, *args, **kwargs):
+        additional_update_fields = []
         if not self.code:
             from corehq.apps.commtrack.util import unicode_slug
             self.code = unicode_slug(self.name)
+            additional_update_fields.append('code')
         if not self.commtrack_enabled:
             self.administrative = True
+            additional_update_fields.append('administrative')
 
         config = stock_level_config_for_domain(self.domain, self.commtrack_enabled)
         if config:
-            self._populate_stock_levels(config)
+            self._populate_stock_levels(config, update_fields=additional_update_fields)
 
         is_not_first_save = self.pk is not None
+        if 'update_fields' in kwargs.keys():
```

Comment:
nit
```suggestion
        if 'update_fields' in kwargs:
```
There are a bunch of other superfluous `.keys()` calls in this PR that could also be changed in the same way.

Reply:
Thanks. See 7d63f0f856fc8bb92521e3efa0d7259458b06bd7